index.js

```diff
@@ -7,256 +7,273 @@ var _ = require('lodash');
+     * @param  {boolean}  useCache          Indicates if cache should be used or not
+     * @callback callback
+     */
+  EchoClient.prototype.queryAnalytics = function queryAnalytics(
+    token,
+    queryOperator,
+    queryParams,
+    useCache,
+    callback
+  ) {
     if (!token) {
-        throw new Error('Missing Persona token');
+      throw new Error('Missing Persona token');
     }
 
     if (!queryOperator) {
-        throw new Error('Missing Analytics queryOperator');
+      throw new Error('Missing Analytics queryOperator');
     }
 
     var validOperators = ['hits', 'average', 'sum', 'max', 'funnel'];
 
     if (validOperators.indexOf(queryOperator) === -1) {
-        throw new Error('Invalid Analytics queryOperator');
+      throw new Error('Invalid Analytics queryOperator');
     }
 
     if (!queryParams) {
-        throw new Error('Missing Analytics queryParams');
+      throw new Error('Missing Analytics queryParams');
     }
 
-    var constructQueryStringResponse = this._queryStringParams(queryParams);
+    var constructQueryStringResponse = queryStringParams(queryParams);
 
     if (!_.isEmpty(constructQueryStringResponse.errors)) {
-        this.error('One or more invalid analytics queryParams where supplied: ' + constructQueryStringResponse.errors.join());
-        throw new Error('Invalid Analytics queryParams');
+      error(
+        'One or more invalid analytics queryParams where supplied',
+        { err: constructQueryStringResponse.errors.join() }
+      );
+
+      throw new Error('Invalid Analytics queryParams');
     }
 
     var requestOptions = {
-        url: this.config.echo_endpoint + '/1/analytics/' + queryOperator + '?' + constructQueryStringResponse.queryString,
-        method: 'GET',
-        headers: {
-            'Content-Type': 'application/json',
-            'Authorization': 'Bearer ' + token
-        }
+      url: config.echo_endpoint + '/1/analytics/' + queryOperator + '?' + constructQueryStringResponse.queryString,
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + token
+      }
     };
 
     if (useCache === false) {
-        requestOptions.headers['cache-control'] = 'none';
+      requestOptions.headers['cache-control'] = 'none';
     }
 
-    this.debug(JSON.stringify(requestOptions));
+    debug('request options', requestOptions);
 
-    var _this = this;
+    request.get(requestOptions, function onResp(err, response, rawBody) {
+      var status = response && response.statusCode
+        ? response.statusCode
+        : 0;
 
-    request.get(requestOptions, function(err, response, body) {
-        if (err) {
-            _this.error('[echoClient] queryAnalytics error: ' + JSON.stringify(err));
-            callback(err);
-        } else {
-            _this._parseJSON(body, callback);
-        }
+      if (err || status / 100 !== 2) {
```

Comment:
won't this fail for status other than 200? is that the intention? i.e. 202 would be 2.02 so would fail.

if you are only interested in 200, why not just specifically test for that?

Reply:
good catch, it is meant to convert the value to a integer